restapi/src/main/java/io/stargate/web/docsapi/service/ReactiveDocumentService.java

```diff
@@ -110,13 +112,46 @@ public ReactiveDocumentService(
     this.configuration = configuration;
   }
 
+  private Single<Integer> determineTtl(
+      DocumentDB db,
+      String namespace,
+      String collection,
+      String documentId,
+      ExecutionContext context) {
+
+    return Single.defer(
+        () -> {
+          authorizeRead(db, namespace, collection);
```

Comment:
I am starting to think if this is needed? Means if you have only write permissions, you can not actually do update or patch with the TTL auto? Is this correct?

Reply:
Oh that's a good point...Yeah maybe this shouldn't be here then (even though it's technically a read, people should be able to update/patch regardless)